pkg/block/fetcher.go

```diff
@@ -612,15 +612,16 @@ const BlockIDLabel = "__block_id"
 
 // Filter filters out blocks that have no labels after relabelling of each block external (Thanos) labels.
 func (f *LabelShardedMetaFilter) Filter(_ context.Context, metas map[ulid.ULID]*metadata.Meta, synced GaugeVec, modified GaugeVec) error {
-	var lbls labels.Labels
+	var b labels.Builder
```

Comment:
I am curious if it is better to just switch to `labels.ScratchBuilder` in this case?
But maybe this is out of the scope of this pr

Reply:
Yeah i would say out of scope; but generally the amount of builders probably could be .. at least pooled or something.